examples/hackernews/README.md

```diff
@@ -20,3 +20,161 @@ transparent to clients, and can be run with:
+   - PostgreSQL database for persistent storage
+   - Stores all application data including posts, users, and votes
+   - Serves as the single source of truth for the system
+
+5. **Reactive Layer (skip_cluster)**
+   - Implements the Skip reactive programming model
+   - Two operational modes:
+     
+     a) **Single Service Mode**
+     - Single Skip service handles all reactive operations
+     - Direct connection to PostgreSQL
+     - Manages both read and write operations
+     
+     b) **Distributed Mode (skip_cluster)**
+     - Leader-Follower architecture:
+       - **Leader**: Handles all write operations and database synchronization
+       - **Followers**: Three instances sharing read operations in round-robin
+     - Followers maintain consistency by syncing with the Leader
+     - Provides horizontal scalability for read operations
+     - Transparent to clients (no client-side changes needed)
+
+6. **Key Architectural Features**
+   - Real-time updates through EventSource/Server-Sent Events
+   - Clear separation of concerns between services
+   - Scalable architecture through distributed Skip services
+   - Configuration-based switching between single and distributed modes
+   - Health-check based service dependencies
+   - Persistent storage with PostgreSQL
+   - Load-balanced read operations in distributed mode
+
+7. **Data Flow**
+   - Client requests enter through HAProxy
+   - Write operations (POST/PUT/DELETE) route to Flask service or Skip Leader
+   - Read operations (GET) can be served by Flask or Skip Followers
+   - Real-time updates flow from Skip services to clients via EventSource
+   - Database serves as the ultimate source of truth
+
+This architecture demonstrates a modern approach to building reactive web applications, combining traditional REST APIs with real-time capabilities, while providing flexibility in deployment through both single-service and distributed configurations.
+
+### Dataflow when posting
+
+```mermaid
+sequenceDiagram
+    %% Title: Post Creation Flow
+    %% Description: Shows the write path when a user creates a new post
+
+    participant Frontend as React Frontend
+    participant HAProxy as HAProxy
+    participant Web as Flask Web Service
+    participant DB as PostgreSQL
+    participant Skip as Skip Service
+
+    Frontend->>HAProxy: POST /api/posts
+    HAProxy->>Web: POST /posts (strips /api/)
+    Web->>DB: INSERT INTO posts
+    Web->>Skip: PATCH /inputs/posts (new post data)
+    Skip->>DB: Sync with new data
+```
+
+### Dataflow for continuous reading
+
+```mermaid
+sequenceDiagram
+    %% Title: Real-time Updates Flow
+    %% Description: Shows how real-time updates are delivered to clients using Server-Sent Events
+
+    participant Frontend as React Frontend
+    participant HAProxy as HAProxy
+    participant Skip as Skip Service
+    participant Web as Flask Web Service
+
+    Frontend->>HAProxy: GET /api/posts (EventSource)
+    HAProxy->>Skip: GET /streams/posts
```

Comment:
This assumes that a stream has already been initiated.
A stream is initiated by the web app (responsible eg. for authorization) on behalf of the user. So the full flow is:
```mermaid
sequenceDiagram
Frontend ->> HAProxy: GET /api/posts (EventSource)
HAProxy ->> Web: GET /api/posts
Web ->> Skip: POST /v1/streams/posts
Skip ->> Web: stream_id
Web ->> Frontend: Redirect to /api/streams/{stream_id}
Frontend ->> HAProxy: GET /api/streams/{stream_id}
HAProxy ->> Skip: GET /v1/streams/{stream_id}
Skip ->> Frontend: SSE
```